### DISCUSSION TITLE
Emoji cut-off

### DISCUSSION THREAD

**Discussion by matthijsgroen** [matteobruni/tsparticles#5402] [Q&A]
I'm using the ghost emoji, which looks different between iPhone and Android. on iPhone it fits within the particle, on android it is cut off on the top, as if it does not fit in the square of the particle. Is there a way to fix this? reducing the size of the emoji inside of the particle, or adding some sort of padding?

**Comment by matteobruni** [comment]
Sorry but I don't have an Android to test it, can you share some code and screenshots?

**Reply by matthijsgroen** [reply]
The game is here: https://matthijsgroen.github.io/block-sort

The code for the ghosts:

```
const ghosts: ISourceOptions = {
  detectRetina: true,
  fpsLimit: 30,
  pauseOnBlur: true,
  particles: {
    number: {
      value: 10,
    },
    move: {
      direction: MoveDirection.top,
      enable: true,
      random: false,
      straight: false,
    },
    opacity: {
      value: { min: 0.1, max: 0.5 },
    },
    size: {
      value: { min: 3, max: 20 },
    },
    shape: {
      type: "emoji",
      options: {
        emoji: {
          font: "sans-serif",
          value: "️👻",
        },
      },
    },

    wobble: {
      distance: 20,
      enable: true,
      speed: {
        min: -5,
        max: 5,
      },
    },
  },
};
```
![signal-2024-10-04-190559_002](https://github.com/user-attachments/assets/ec9d380a-cc33-435e-80be-28e171b3d4e1)

**Reply by matteobruni** [reply]
I've added a `padding` options to 3.6.0, I need to check some other things, but hope to release it in few days

**Reply by matteobruni** [reply]
Can you try 3.6.0-beta.0?

https://github.com/tsparticles/tsparticles/blob/09ffcbf5d5210c3da84fbb7ed8ef9a5d83c9cc9e/utils/configs/src/s/shapeEmojiPadding.ts#L34-L42

This is how the new padding option works

**Reply by matthijsgroen** [reply]
Yes I will try it out tonight. Thanks for the quick update! I also got back from users that the performance was stuttering on their phones. So I had to change the ghosts/ snow effect to a small self-build particle system that uses CSS animations. But I want to use particles in other places too so I will check out the updates.

Do you have any idea why mobile performance is stuttering ? it is super smooth on desktop.

By the way this is my small CSS particles I set up now: https://github.com/matthijsgroen/block-sort/blob/main/src/ui/Background/CSSParticles.tsx It is mainly good in particles that are repeating, like the ghosts and the snow, but not good for emitter bursts like confetti. So I do want to embed your engine for those things :-)

**Reply by matteobruni** [reply]
I have an idea, I found a memory leak that is fixed in that version too. Your config should be fine
